### PR TITLE
refactor(packagist): Type checked Packagist lookups

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

This time, don't be so strict on input side

## Context

- Fixes: #19808

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/packagist/schema.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    .catch(null),
+  time: z.string().nullable().catch(null),
+});
+
+export const ComposerReleaseArray = z
+  .array(ComposerRelease.nullable().catch(null))
+  .transform((xs) =>
+    xs.filter((x): x is z.infer<typeof ComposerRelease> => x !== null)
+  );
+export type ComposerReleaseArray = z.infer<typeof ComposerReleaseArray>;
```
Comment (viceice) [inline lib/modules/datasource/packagist/schema.ts:25]: are you sure we want the const and type having same name? 🤔

Comment (zharinov) [reply lib/modules/datasource/packagist/schema.ts:25]: What downsides you can see? As far as I can see, they reside in different namespaces that are quite orthogonal. What can go wrong?

Comment (viceice) [reply lib/modules/datasource/packagist/schema.ts:25]: won't typescript merge both types, which is not the behavior we want? 🤔

Comment (zharinov) [reply lib/modules/datasource/packagist/schema.ts:25]: I was okay using same name, my dev experience was that it knows what does it mean in every context

Comment (viceice) [reply lib/modules/datasource/packagist/schema.ts:25]: ok, still looks strange to me. 🤔

**Review by viceice — APPROVED** [review]

**Comment by bobvandevijver** [comment]
Great, can confirm this works just fine :)

**Pull request merged**